SURF-1188: Add example and note about importing WITH and Union branches (#1604)
neo4j/docs-cypher: modules/ROOT/pages/subqueries/call-subquery.adoc

> * The importing `WITH` clause must the first clause of a subquery (or the second clause, if directly following a `USE` clause).

Reviewer: ```suggestion
* The importing `WITH` clause must be the first clause of a subquery (or the second clause, if directly following a `USE` clause).
```

* It is not possible to follow an importing `WITH` clause with any of the following clauses: `DISTINCT`, `ORDER BY`, `WHERE`, `SKIP`, and `LIMIT`.

> * When using a `UNION [ALL]` directly within the `CALL` any imported variables need to be imported per `UNION` branch.

Reviewer: ```suggestion
* When using a `UNION [ALL]` directly within the `CALL` any imported variables must be imported per `UNION` branch.
```

|===

> Attempting to use an outer variable only imported into the first `UNION` branch in the second will throw an error:

Reviewer: ```suggestion
Attempting to use an outer variable only imported into the first `UNION` branch in the second throws an error:
```
